src/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -441,4 +441,7 @@
   <data name="InvalidDuplicatePropertyNameHandling" xml:space="preserve">
     <value>The DuplicatePropertyNameHandling enum must be set to one of the supported values.</value>
   </data>
+  <data name="InvalidLeadingZerosInNumber" xml:space="preserve">
+    <value>Invalid leading zeros before '{0}'. Expected '.', 'e', 'E' or a delimiter after the zero.</value>
```

Comment:
I think this exception message makes sense, generally. I don't think we ever reach more than one leading zero, so this should be singular:
```suggestion
    <value>Invalid leading zero before '{0}'. Expected '.', 'e', 'E' or a delimiter after the zero.</value>
```


Just as a side note, I was thinking, if a JSON payload contains "01", did the user intend/expect "1"? If so, the error message saying we expected a '.', 'e', 'E', or a delimiter may not be useful, since the problem isn't with the "1", it's with the extra zero. So, maybe we just state `Invalid leading zeros before '{0}'.`

Reply:
>  I don't think we ever reach more than one leading zero

It would be good to add a test for something like "001" and "00h" to verify that and make sure that behavior holds. We only have a case with a single leading zero:
https://github.com/dotnet/corefx/blob/9be99cb634ffe1b8dc60e4b3d8c01026347a0283/src/System.Text.Json/tests/Utf8JsonReaderTests.cs#L2040